Do not try to install GAPID APK when Android version is too low

The logic in `core/os/android/adb/device.go:newDevice()` should stop early and not trigger the DeviceInfoProviders when Android version is below requirements.

Currently, it tries to run the gapid APK device info, which crashes on old versions of Android (see crash trace below). Rather than trying to fix that crash, we should stop early and not even try to gather detailed device info when the Android version is lower than M (as advertised on https://gapid.dev/requirements/ ).

Crash trace:
```
E/gapid-pkginfo(19548): Error occurred
E/gapid-pkginfo(19548): java.util.concurrent.ExecutionException: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "__register_atfork" referenced by "libdeviceinfo.so"...
E/gapid-pkginfo(19548): 	at java.util.concurrent.FutureTask.report(FutureTask.java:93)
E/gapid-pkginfo(19548): 	at java.util.concurrent.FutureTask.get(FutureTask.java:163)
E/gapid-pkginfo(19548): 	at com.google.android.gapid.SocketWriter.connectAndWrite(SocketWriter.java:46)
E/gapid-pkginfo(19548): 	at com.google.android.gapid.DeviceInfoService.handleSendDeviceInfo(DeviceInfoService.java:90)
E/gapid-pkginfo(19548): 	at com.google.android.gapid.DeviceInfoService.onHandleIntent(DeviceInfoService.java:71)
E/gapid-pkginfo(19548): 	at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
E/gapid-pkginfo(19548): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/gapid-pkginfo(19548): 	at android.os.Looper.loop(Looper.java:145)
E/gapid-pkginfo(19548): 	at android.os.HandlerThread.run(HandlerThread.java:61)
E/gapid-pkginfo(19548): Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "__register_atfork" referenced by "libdeviceinfo.so"...
E/gapid-pkginfo(19548): 	at java.lang.Runtime.loadLibrary(Runtime.java:371)
E/gapid-pkginfo(19548): 	at java.lang.System.loadLibrary(System.java:989)
E/gapid-pkginfo(19548): 	at com.google.android.gapid.DeviceInfoService$1.call(DeviceInfoService.java:83)
E/gapid-pkginfo(19548): 	at com.google.android.gapid.DeviceInfoService$1.call(DeviceInfoService.java:80)
E/gapid-pkginfo(19548): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
E/gapid-pkginfo(19548): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/gapid-pkginfo(19548): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/gapid-pkginfo(19548): 	at java.lang.Thread.run(Thread.java:818)
```

--- a/core/os/android/adb/device.go
+++ b/core/os/android/adb/device.go
@@ -167,11 +167,14 @@ func newDevice(ctx context.Context, serial string, status bind.Status) (*binding
 		}
 	}
 
-	devInfoProvidersMutex.Lock()
-	defer devInfoProvidersMutex.Unlock()
-	for _, f := range devInfoProviders {
-		if err := f(ctx, d); err != nil {
-			return nil, err
+	// Run device info providers only if the API is supported
+	if d.To.Configuration.OS != nil && d.To.Configuration.OS.APIVersion >= device.AndroidMinimalSupportedAPIVersion {
+		devInfoProvidersMutex.Lock()
+		defer devInfoProvidersMutex.Unlock()
+		for _, f := range devInfoProviders {
+			if err := f(ctx, d); err != nil {
+				return nil, err
+			}
 		}
 	}
 

--- a/core/os/device/android.go
+++ b/core/os/device/android.go
@@ -16,6 +16,8 @@ package device
 
 import "fmt"
 
+const AndroidMinimalSupportedAPIVersion = 23
+
 // AndroidOS returns the full OS structure for the supplied android os version.
 func AndroidOS(major, minor, point int32) *OS {
 	os := &OS{

